presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java

```diff
@@ -170,6 +171,15 @@ public static Slice toIEEE754Binary32(@SqlType(StandardTypes.REAL) long value)
         return slice;
     }
 
+    @Description("decode value as a single-precision floating-point according to IEEE 754 format")
+    @ScalarFunction("from_ieee754_32")
+    @SqlType(StandardTypes.REAL)
+    public static long fromIEEE754Binary32(@SqlType(StandardTypes.VARBINARY) Slice slice)
+    {
+        ByteBuffer buffer = ByteBuffer.wrap(slice.getBytes());
```

Comment:
We should be able to use `slice.getInt(0)` here.

Reply:
Add checks for length (see `StringFunctions` for more examples):
```java
checkCondition(slice.length() == 4, INVALID_FUNCTION_ARGUMENT, "Input floating-point value must be exactly 4 bytes long");
```
Then add tests for this using `assertInvalidFunction()` (see `TestStringFunctions` for examples)

Reply:
We may not be able to use `slice.getInt(0)` here..

```
java.lang.AssertionError: expected [1.0] but found [4.6006E-41]
	at com.facebook.presto.operator.scalar.TestVarbinaryFunctions.testFromIEEE754Binary32(TestVarbinaryFunctions.java:208)
```

It returns a different result from the result of `ByteBuffer#getInt`.  Do you have any idea?